In [1]:
import pandas as pd
import numpy as np

income_df = pd.read_excel('ny_zip_income.xls',header=3,usecols=[0,1,2,16,17,18])

### Explore Income dataset

In [2]:
income_df.head().T

0    1            2  \
ZIP\ncode [1]                                  NaN  NaN            0   
Size of adjusted gross income                  NaN  NaN        Total   
Number of returns                              NaN   -1  9.46806e+06   
Adjusted gross income (AGI) [4]                NaN  -15  7.84411e+08   
Total income                     Number of returns  -16      9468050   
Unnamed: 18                                 Amount  -17    795119743   

                                                3                      4  
ZIP\ncode [1]                                   0                      0  
Size of adjusted gross income    $1 under $25,000  $25,000 under $50,000  
Number of returns                     3.44531e+06            2.12396e+06  
Adjusted gross income (AGI) [4]       4.19902e+07            7.72745e+07  
Total income                              3445290                2123960  
Unnamed: 18                              42897756               78326525

### Data Cleaning - Income dataset

In [3]:
#rename columns
income_df = income_df.rename(columns={income_df.columns[0]: "zip_code", 
                                      income_df.columns[1]: "agi_bucket", 
                                      income_df.columns[2]: "num_returns", 
                                      income_df.columns[3]: "agi", 
                                      income_df.columns[4]: "tot_inc_num_returns", 
                                      income_df.columns[5]: "tot_inc_amnt"})
#only take the meaningful rows, avoiding totals
income_df = income_df.iloc[3:12329]

In [4]:
#change dollars (in thousands) columns to exact dollars
income_df['agi'] = income_df['agi']*1000
income_df['tot_inc_amnt'] = income_df['tot_inc_amnt']*1000

In [5]:
#drop rows that don't contain an agi_bucket value, they are pointless
income_df.dropna(subset=['agi_bucket'],inplace=True)

In [6]:
income_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9246 entries, 3 to 12328
Data columns (total 6 columns):
zip_code               9246 non-null object
agi_bucket             9246 non-null object
num_returns            9246 non-null float64
agi                    9246 non-null float64
tot_inc_num_returns    9246 non-null object
tot_inc_amnt           9246 non-null object
dtypes: float64(2), object(4)
memory usage: 505.6+ KB


In [7]:
#remove Zipcodes 0 & 99999
income_df = income_df.iloc[6:-6] 

In [8]:
#reset dataframe index
income_df.reset_index(drop=True,inplace=True)

In [9]:
income_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9234 entries, 0 to 9233
Data columns (total 6 columns):
zip_code               9234 non-null object
agi_bucket             9234 non-null object
num_returns            9234 non-null float64
agi                    9234 non-null float64
tot_inc_num_returns    9234 non-null object
tot_inc_amnt           9234 non-null object
dtypes: float64(2), object(4)
memory usage: 432.9+ KB


In [10]:
income_df.head()

zip_code               agi_bucket  num_returns          agi  \
0    10001         $1 under $25,000       3740.0   43182000.0   
1    10001    $25,000 under $50,000       2500.0   92133000.0   
2    10001    $50,000 under $75,000       1950.0  120922000.0   
3    10001   $75,000 under $100,000       1410.0  121880000.0   
4    10001  $100,000 under $200,000       2580.0  361453000.0   

  tot_inc_num_returns tot_inc_amnt  
0                3740     44849000  
1                2500     94324000  
2                1950    123197000  
3                1410    123605000  
4                2580    366195000

In [11]:
#show rows that have a num_returns discrepancy
income_df[income_df['tot_inc_num_returns'] != income_df['num_returns']]

zip_code              agi_bucket  num_returns          agi  \
1584    11355        $1 under $25,000      34950.0  401891000.0   
7067    13833        $200,000 or more         30.0   12181000.0   
9105    14874  $75,000 under $100,000         20.0    2656000.0   

     tot_inc_num_returns tot_inc_amnt  
1584               34940    412155000  
7067                  20     12181000  
9105                  30      2656000

In [12]:
#convert zip_code column to integer datatype
income_df['zip_code'].astype('int64')

0       10001
1       10001
2       10001
3       10001
4       10001
5       10001
6       10002
7       10002
8       10002
9       10002
10      10002
11      10002
12      10003
13      10003
14      10003
15      10003
16      10003
17      10003
18      10004
19      10004
20      10004
21      10004
22      10004
23      10004
24      10005
25      10005
26      10005
27      10005
28      10005
29      10005
        ...  
9204    14898
9205    14898
9206    14898
9207    14898
9208    14898
9209    14898
9210    14901
9211    14901
9212    14901
9213    14901
9214    14901
9215    14901
9216    14903
9217    14903
9218    14903
9219    14903
9220    14903
9221    14903
9222    14904
9223    14904
9224    14904
9225    14904
9226    14904
9227    14904
9228    14905
9229    14905
9230    14905
9231    14905
9232    14905
9233    14905
Name: zip_code, Length: 9234, dtype: int64

### Filter New York State zip codes down to only New York City zip codes

In [13]:
#load data
nyc_zips_df = pd.read_csv('zip_borough.csv',usecols=[0]) #SOURCE: https://www.kaggle.com/kimjinyoung/nyc-borough-zip/downloads/nyc-borough-zip.zip/2


In [14]:
nyc_zips_df.head()

zip
0  10001
1  10002
2  10003
3  10004
4  10005

In [15]:
#reset the income_df to only include NYC zip codes as seen in the nyc_zips_df
income_df = income_df[income_df['zip_code'].isin(nyc_zips_df['zip'])].reset_index(drop=True)

In [16]:
#convert numerical columns to float datatype
income_df['tot_inc_num_returns'] = income_df['tot_inc_num_returns'].astype('float64')
income_df['tot_inc_amnt'] = income_df['tot_inc_amnt'].astype('float64')

In [17]:
income_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1062 entries, 0 to 1061
Data columns (total 6 columns):
zip_code               1062 non-null object
agi_bucket             1062 non-null object
num_returns            1062 non-null float64
agi                    1062 non-null float64
tot_inc_num_returns    1062 non-null float64
tot_inc_amnt           1062 non-null float64
dtypes: float64(4), object(2)
memory usage: 49.9+ KB


In [24]:
income_df.describe()

num_returns           agi  tot_inc_num_returns  tot_inc_amnt
count   1062.000000  1.062000e+03          1062.000000  1.062000e+03
mean    3785.969868  3.079024e+08          3785.960452  3.118991e+08
std     4365.667804  8.275105e+08          4365.600534  8.383446e+08
min        0.000000  0.000000e+00             0.000000  0.000000e+00
25%     1070.000000  8.402525e+07          1070.000000  8.529400e+07
50%     2400.000000  1.599225e+08          2400.000000  1.615430e+08
75%     4897.500000  2.712135e+08          4897.500000  2.753948e+08
max    36120.000000  9.954884e+09         36120.000000  1.008504e+10

### Explore Parks Dataset

In [18]:
parks_df = pd.read_json('DPR_Parks_001.json')

In [27]:
parks_df.head(30)

Location  \
0          Franklin St. bet. Milton St. and Noble St.   
1         E. 38 St., E. 38 St. bet. Ave. I and Ave. J   
2                            Prospect Park W., 15 St.   
3              Broadway, Stuyvesant Ave., Vernon Ave.   
4             Cropsey Ave. bet. 21 Ave. and Bay Pkwy.   
5   Blake Ave., Dumont Ave., Livonia Ave. bet. Str...   
6   Washington Ave., Flatbush Ave. bet. Eastern Pk...   
7   St. Mark's Ave., Park Pl. bet. Brooklyn Ave. a...   
8   Irving Ave., Kinickerbocker Ave. bet. Starr St...   
9   Knickerbocker Ave. bet. Woodbine St. and Putna...   
10  Seaview Ave. bet. Paerdegat Basin and E. 93 St...   
11  Court St., Smith St., bet. Carroll St. and Pre...   
12  Nassau St., Park Ave., bet. Navy St. and N. El...   
13  Lynch St., Middleton St. bet. Lee Ave. and Bed...   
14  Lafayette Ave. bet. Malcolm X Blvd. and Patch ...   
15  Maspeth Ave., Sharon St. bet. Olive St. and Mo...   
16               Carlton Ave., Fulton St., Green Ave.   
17  Hegeman Ave., New Lots Ave. bet. Watkins St. a...   
18  86 St., Belt Pkwy. bet. Bay 8 St., 14 Ave., an...   
19  Eastern Pkwy. bet. Grand Army Plaza and Ralph ...   
20    Eastern Pkwy. bet. Ralph Ave. and Bushwick Ave.   
21       Monitor St. bet. Engert Ave. and Meeker Ave.   
22  Myrtle Ave., De Kalb Ave. bet. Washington Park...   
23                             5 Ave., 4 Ave., 94 St.   
24  101 St., Shore Pkwy. bet. 4 Ave. and Ft. Hamil...   
25  Clark St. at Columbia Heights and the Brooklyn...   
26        Bushwick Ave., Myrtle Ave., Willoughby Ave.   
27   Chauncey St. bet. Lewis Ave. and Stuyvesant Ave.   
28     Flatbush Ave., Eastern Pkwy. and Prospect Park   
29             Bedford Ave., Rodgers Ave., Bergen St.   

                         Name Prop_ID                                Zip  
0         American Playground    B001                              11222  
1              Amersfort Park    B002                              11210  
2     Bartel-Pritchard Square    B003                              11215  
3              Beattie Square    B006                              11221  
4            Bensonhurst Park    B007                              11214  
5             Betsy Head Park    B008                              11212  
6     Brooklyn Botanic Garden    B010                              11238  
7                 Brower Park    B012                              11213  
8        Maria Hernandez Park    B016                              11237  
9         Bushwick Playground    B017                              11237  
10              Canarsie Park    B018                       11236, 11239  
11               Carroll Park    B019                              11231  
12       Commodore Barry Park    B021                              11201  
13       Middleton Playground    B022                              11206  
14       Lafayette Playground    B023                              11221  
15                Cooper Park    B025                              11211  
16           Cuyler Gore Park    B026                              11238  
17          Veterans Triangle    B027                              11212  
18           Dyker Beach Park    B028                       11209, 11228  
19            Eastern Parkway    B029  11213, 11216, 11225, 11233, 11238  
20  Eastern Parkway Extension    B030                       11207, 11233  
21          Fidelity Triangle    B031                              11222  
22           Fort Greene Park    B032                       11201, 11205  
23     Fort Hamilton Triangle    B033                              11209  
24       John Paul Jones Park    B035                              11209  
25         Fort Stirling Park    B036                              11201  
26           Freedom Triangle    B037                              11221  
27                Fulton Park    B038                              11233  
28           Grand Army Plaza    B040                11215, 11217, 11238  
29                 Grant Go

In [20]:
parks_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1675 entries, 0 to 1674
Data columns (total 4 columns):
Location    1675 non-null object
Name        1675 non-null object
Prop_ID     1675 non-null object
Zip         1569 non-null object
dtypes: object(4)
memory usage: 52.4+ KB


In [63]:
parks_df.nunique()

Location    1669
Name        1532
Prop_ID     1675
Zip          278
dtype: int64

In [66]:
#explore duplicate park names
parks_df[parks_df.duplicated(subset=['Name'],keep=False)].sort_values(by=['Name'])

Location  \
1376  Dr. MLK Jr. Blvd. bet. W. Tremont Ave. and E. ...   
1375  Aqueduct Ave. W. bet. W. Fordham Rd. and W. Ki...   
713   bet. Hudson River and South End Ave. S/o Liber...   
712       Hudson River, Liberty St. West St, and Pier A   
312       Belmont Ave. between Forbell St. and Drew St.   
1471             E 182 St bet. Belmont Av and Croton Av   
1460  Harlem River bet. W 175 St and Alexander Hamil...   
220   Prospect St., York St. bet. Jay St. and Bridge...   
550               Broadway, Columbus Circle To W 110 St   
552                  Broadway, W. 135 St. To W. 156 St.   
553                      Broadway, W 156 St To W 168 St   
551                       Broadway, W 110 StTo W 122 St   
202                   BQE bet. Remsen St. and Grace Ct.   
210        BQE over-hang bet. Remsen St. and Orange St.   
209               BQE bet. Pineapple St. and Orange St.   
208     Pineapple St. bet. Columbia Heights and the BQE   
207                BQE bet. Clark St. and Pineapple St.   
206              BQE bet. Clark St. and Pierrepoint St.   
205                                 BQE at Montague St.   
204             Montague St. bet. Montague Ter. and BQE   
203                BQE bet. Remsen St. and Montague St.   
201                BQE bet. Joralemon St. and Grace Ct.   
337       Flushing Ave. bet.. Beaver St. and Garden St.   
9     Knickerbocker Ave. bet. Woodbine St. and Putna...   
1351                       Carlton Blvd. and Crown Ave.   
1269          Foste Rd., Carlton Ave., Drumgoole Rd. W.   
322                     Lafayette Ave. and Classon Ave.   
224   Kent Ave. to Flushing Ave., bet. Williamsburg ...   
477          Baxter St, Mulberry St, Bayard St and Park   
84    Adam St., Court St., Cadman Plaza West bet. Jo...   
...                                                 ...   
923     Rockaway Boardwalk bet. Crest Rd. and B. 17 St.   
282                          Sherman St. at Ocean Pkwy.   
218              Pearl St. at York St. and Manh. Bridge   
567                               E. 93 St. and FDR Dr.   
521                E. 59 St. To E. 97 St. and Park Ave.   
520                      E 34 St To E 39 St and Park Av   
519                      E 46 St To E 59 St and Park Av   
1408            Coutlandt Av bet. E 161 St and E 162 St   
1114                  129 Ave. bet. 172 St. and 176 St.   
532         Riverside Dr to Henry Hudson Pkwy, W 153 St   
531   Riverside Dr. to Hudson River, W. 72 St.to St ...   
927       Atlantic Ocean bet. B. 149 St. and B. 126 St.   
827   Ocean Promenade, bet. Beach 126 St. and Beach ...   
280   N/B Prospect Exwy. bet. Seeley St. and Greenwo...   
846                    Hoyt Ave. bet. 19 St. and 21 St.   
1063  58 Ave., 73 St., Queens - Mid-Town Exwy. Sr. R...   
1065                                  74 St. at 57 Ave.   
386            Shore Pkwy and Seaview Ave., Schenk Ave.   
133               Belt Pkwy., Fresh Creek, Seaview Ave.   
132   Fountain Ave., 75 St. bet. Belt Pkwy., Flatlan...   
1421       St Mary's St bet. St Ann's Av and Jackson Av   
91         Smith St. bet. Nelson St. and Huntington St.   
90             Smith St. bet. Luquer St. and Nelson St.   
122             New Lots Ave., Dumont Ave., Atkins Ave.   
1426               E 163 St, Washington Av and Brook Av   
248                    Ralph Ave., Ave. T and E. 61 St.   
831                      National St., 42 Ave., 102 St.   
1191               FL Blvd & Sunrise Hwy/So. Conduit Av   
81                  5 St., 4 St. bet. 4 Ave. and 5 Ave.   
1636            E 183 St bet. Washington Av and Park Av   

                            Name Prop_ID  \
1376               Aqueduct Walk   X001A   
1375               Aqueduct Walk    X001   
713            Battery Park City   M283A   
712            Battery Park City    M283   
312           Belmont Playground    B283   
1471          Belmont Playground    X108   
1460                 Bridge Park    X094   
220                  Bridge Park  B2

- Insight: park names may span multiple Prop_ID's and multiple Zip Codes
- All 1,675 Prop_IDs are unique to each row - Keep all prop_id's

In [23]:
parks_df.describe()

Location  Name Prop_ID    Zip
count                               1675  1675    1675   1569
unique                              1669  1532    1675    278
top     Hoyt Ave. bet. 21 St. and 23 St.  Park    B280  11211
freq                                   2    36       1     41

### Explore Playgrounds dataset

In [28]:
playgrounds_df = pd.read_json('DPR_Playgrounds_001.json')

In [31]:
playgrounds_df.head(30)

Accessible Adaptive_Swing  Level  \
0           Y              N    4.0   
1           Y              N    4.0   
2           Y              N    1.0   
3           Y              N    2.0   
4           Y              Y    2.0   
5           Y              N    3.0   
6           N           None    NaN   
7           Y              N    4.0   
8           Y              N    4.0   
9           Y              Y    3.0   
10          Y              N    4.0   
11          Y              N    4.0   
12          Y              N    4.0   
13          Y              N    4.0   
14          Y              N    4.0   
15          Y              N    4.0   
16          Y              N    4.0   
17          Y              N    2.0   
18          Y              N    2.0   
19          Y              N    2.0   
20          Y              N    3.0   
21          Y              Y    4.0   
22          Y              N    4.0   
23          Y              N    4.0   
24          Y              N    2.0   
25          N           None    NaN   
26          N           None    NaN   
27          Y              Y    4.0   
28          Y              Y    3.0   
29          Y              N    4.0   

                                             Location  \
0                     Noble, Franklin, Milton Streets   
1                           BAY PKWY & CROPSEY AVENUE   
2                             DUMONT AVE & BRISTOL ST   
3                          BROOKLYN AVE & PROSPECT PL   
4                          BROOKLYN AVE & PROSPECT PL   
5                           KNICKERBOCKER ST ENTRANCE   
6                                 IRVING AVE ENTRANCE   
7             Knickerbocker, Putnam Aves, Woodbine St   
8                               E. 92 ST, SEAVIEW AVE   
9                Carroll, Court, President, Smith Sts   
10        Park Ave, Navy St, Flushing Ave, N Eliot Pl   
11                      Lynch, Middleton Sts, Lee Ave   
12                  Lafayette Ave, E/o Malcolm X Blvd   
13                      SHARON & OLIVE STS,MORGAN AVE   
14                 Fulton, Cumberland Sts, Greene Ave   
15                                    CROPSEY & BAY 8   
16                                     14 AVE & 86 ST   
17                      MYRTLE AVE & ST EDWARDS PLAZA   
18                         DEKALB AVE & CUMBERLAND ST   
19                                     18 AVE & 56 ST   
20                   Commercial, Franklin, Dupont Sts   
21  Knickerbocker, Wilson Aves, Weirfield, Halsey Sts   
22             Congress, Columbia, Pacific, Hicks Sts   
23                              Ave S, E 14 & E 8 Sts   
24  4TH AVE TO CS ADJ 5TH AVE, BTWN 66TH AND 67TH ...   
25  7TH AV (EAST SIDE OF HIGHWAY) TO 8TH AVE, 66TH...   
26  8TH AVE TO FT HAMILTON PKWY, BTWN 66TH AND 67T...   
27                          ROCHESTER AVE, CARROLL ST   
28         Montrose Ave, Boerum, Lorimer, Leonard Sts   
29            Dumont, Blake, Miller Aves, Bradford St   

                                                 Name Playground_ID Prop_ID  \
0                                 American Playground          B001    B001   
1                                    Bensonhurst Park       B007-01    B007   
2                      Betsy Head Memorial Playground       B008-03    B008   
3                                         Brower Park       B012-02    B012   
4                                         Brower Park       B012-03    B012   
5                                Maria Hernandez Park       B016-01    B016   
6                                Maria Hernandez Park          B016    B016   
7                                 Bushwick Playground          B017    B017   
8                                 Dinapoli Playground       B018-01    B018   
9                                        Carroll Park          B019    B019   
10                               Commodore Barry Park          B021    B021   
11                               Middleton Playground          B022    B022  

In [32]:
playgrounds_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1269 entries, 0 to 1268
Data columns (total 11 columns):
Accessible        1266 non-null object
Adaptive_Swing    1173 non-null object
Level             1175 non-null float64
Location          1266 non-null object
Name              1269 non-null object
Playground_ID     1021 non-null object
Prop_ID           1269 non-null object
School_ID         234 non-null object
Status            234 non-null object
lat               1245 non-null float64
lon               1245 non-null float64
dtypes: float64(3), object(8)
memory usage: 109.1+ KB


In [33]:
playgrounds_df.describe()

Level          lat          lon
count  1175.000000  1245.000000  1245.000000
mean      3.571064    40.691978   -73.923549
std       0.792757     1.157509     0.090063
min       1.000000     0.000000   -74.244900
25%       3.000000    40.663300   -73.969800
50%       4.000000    40.717400   -73.925600
75%       4.000000    40.800700   -73.872000
max       4.000000    40.902500   -73.709600

In [59]:
print(playgrounds_df[~playgrounds_df['School_ID'].isna()].shape) #shape of non-null School_ID dataframe
print(playgrounds_df[playgrounds_df['School_ID'].isna()].shape) #shape of null School_ID dataframe
print(playgrounds_df.shape) #shape of dataframe

(234, 11)
(1035, 11)
(1269, 11)


In [60]:
#remove school playgrounds from playgrounds_df
playgrounds_df = playgrounds_df[playgrounds_df['School_ID'].isna()]

In [62]:
playgrounds_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1035 entries, 0 to 1268
Data columns (total 11 columns):
Accessible        1032 non-null object
Adaptive_Swing    939 non-null object
Level             941 non-null float64
Location          1032 non-null object
Name              1035 non-null object
Playground_ID     1021 non-null object
Prop_ID           1035 non-null object
School_ID         0 non-null object
Status            0 non-null object
lat               1018 non-null float64
lon               1018 non-null float64
dtypes: float64(3), object(8)
memory usage: 97.0+ KB


In [68]:
playgrounds_df.nunique()

Accessible           2
Adaptive_Swing       3
Level                4
Location          1027
Name               982
Playground_ID     1017
Prop_ID            865
School_ID            0
Status               0
lat                866
lon                835
dtype: int64

In [69]:
#explore duplicate Playground_ID's
playgrounds_df[playgrounds_df.duplicated(subset=['Playground_ID'],keep=False)].sort_values(by=['Playground_ID'])

Accessible Adaptive_Swing  Level  \
36            Y              N    2.0   
37            Y              N    2.0   
71            Y              N    4.0   
996           Y              N    4.0   
572           Y              Y    3.0   
574           N           None    NaN   
590           Y              N    4.0   
591           Y              N    4.0   
1233          Y              N    4.0   
1237          Y              Y    3.0   
1239          Y           None    NaN   
1240          Y              Y    1.0   
1242          Y              Y    3.0   
1243          Y              N    4.0   
1244          Y              N    4.0   
1245          Y              Y    3.0   
1246          Y              N    4.0   
1247          Y              N    4.0   
1248          Y              Y    4.0   
1249          Y              N    4.0   
1250          Y              N    2.0   
1262          Y              N    4.0   

                                               Location  \
36                67 ST & COLONIAL RD, NORTH OF B066-02   
37         NR ENT @ COLONIAL RD/67 ST, SOUTH OF B066-01   
71                              3 To 4 Sts, 4 To 5 Aves   
996                                     Washington Park   
572                           VERNON BLVD, 40 & 41 AVES   
574                                                       
590                                    225 St & 104 Ave   
591                                    224 ST & 101 AVE   
1233     At the corner of Cleveland and Mansion Avenues   
1237  Boardwalk between Beach 29th and Beach 30th St...   
1239           Kent Ave. between N. 9 St. and N. 12 St.   
1240                           GERRITSEN AVE & SEBA AVE   
1242    Pier 1 - at Old Fulton Street and Furman Street   
1243                       Pier 5 - at Joralemon Street   
1244  Pier 6 - between Atlantic Avenue and Joralemon...   
1245      Pier 6 - at Atlantic Avenue and Furman Street   
1246  Pier 6 - between Atlantic Avenue and Joralemon...   
1247  Pier 6 - between Atlantic Avenue and Joralemon...   
1248                                             Pier 2   
1249  Pier 6 - between Atlantic Avenue and Joralemon...   
1250                                               None   
1262                           51st Ave. & Jackson Ave.   

                              Name Playground_ID Prop_ID School_ID Status  \
36                  Owls Head Park          B066    B066      None   None   
37                  Owls Head Park          B066    B066      None   None   
71                 Washington Park          B111    B111      None   None   
996                Washington Park          B111    B111      None   None   
572              Vernon Playground       Q104-01    Q104      None   None   
574              VERNON PLAYGROUND       Q104-01    Q104      None   None   
590        Pat Williams Playground          Q133    Q133      None   None   
591        Pat Williams Playground          Q133    Q133      None   None   
1233  Seaside Wildlife Nature Park          None    R145      None   None   
1237  Beach 30th Street Playground          None    Q162      None   None   
1239           Bushwick Inlet Park          None    B529      None   None   
1240               Seba Playground          None    B057      None   None   
1242             Pier 1 Playground          None    B431      None   None   
1243             Pier 5 Playground          None    B431      None   None   
1244                      Waterlab          None    B431      None   None   
1245                  Swing Valley          None    B431      None   None   
1246               Sandbox Village          None    B431      None   None   
1247                Slide Mountain          None    B431      None   None   
1248                 Pier 2 Swings          None    B431      None   None   
1249                  Marsh Garden          None    B431      None   None   
1250       Pearl Street Playground          None    M378      None   None   
1262     

- Some Prop_IDs may have multiple playgrounds - use a left join where all playgrounds remain in the set, and relevant zip codes are added to each

In [ ]:
#Impute Prop_ID as Playground ID

### Add Zip Codes from Parks dataset to Playground dataset using LEFT JOIN (playground = left df, parks = right df) on Prop_ID

In [ ]:
playgrounds_df.join(parks_df[['']])

### Join Parks & Playground datasets with Income by Zip Code dataset

In [ ]:


nyc_zips_df = pd.read_csv('zip_borough.csv') 